packages/@aws-cdk/aws-efs/lib/efs-file-system.ts

```diff
@@ -0,0 +1,260 @@
+ */
+export interface EfsFileSystemProps extends FileSystemProps {
+    /**
+     * Defines if the data at rest in the file system is encrypted or not.
+     *
+     * @default - false
+     */
+    readonly encrypted?: boolean;
+
+    /**
+     * The KMS key used for encryption. This is required to encrypt the data at rest if @encrypted is set to true.
+     *
+     * @default - if @encrypted is true, the default key for EFS (/aws/elasticfilesystem) is used
+     */
+    readonly kmsKey?: kms.IKey;
+
+    /**
+     * The key value pair added to the File system.
+     *
+     * @default - no tags will be added
+     */
+    readonly fileSystemTags?: Tag[];
+
+    /**
+     * A policy used by EFS lifecycle management to transition files to the Infrequent Access (IA) storage class.
+     *
+     * @default - none
+     */
+    readonly lifecyclePolicy?: EfsLifecyclePolicyProperty;
+
+    /**
+     * Enum to mention the performance mode of the file system.
+     *
+     * @default - GENERAL_PURPOSE
+     */
+    readonly performanceMode?: EfsPerformanceMode;
+
+    /**
+     * Enum to mention the throughput mode of the file system.
+     *
+     * @default - BURSTING
+     */
+    readonly throughputMode?: EfsThroughputMode;
+
+    /**
+     * Provisioned throughput for the file system. This is a required property if the throughput mode is set to PROVISIONED.
+     * Valid values are 1-1024.
+     *
+     * @default - None, errors out
+     */
+    readonly provisionedThroughputInMibps?: number;
+}
+
+/**
+ *  A new or imported EFS File System.
+ */
+abstract class EFSFileSystemBase extends Resource implements IFileSystem {
+
+    /**
+     * The security groups/rules used to allow network connections to the file system.
+     */
+    public abstract readonly connections: ec2.Connections;
+
+    /**
+     * @attribute
+     */
+    public abstract readonly fileSystemID: string;
+}
+
+/**
+ * Properties that describe an existing EFS file system.
+ */
+export interface EfsFileSystemAttributes {
+    /**
+     * The security group of the file system
+     */
+    readonly securityGroup: ec2.ISecurityGroup;
+
+    /**
+     * The File System's ID.
+     */
+    readonly fileSystemID: string;
+}
+
+/**
+ * The Elastic File System implementation of IFileSystem.
+ * It creates a new, empty file system in Amazon Elastic File System (Amazon EFS).
+ * It also creates mount target (AWS::EFS::MountTarget) implicitly to mount the
+ * EFS file system on an Amazon Elastic Compute Cloud (Amazon EC2) instance or another resource.
+ *
+ * @see https://docs.aws.amazon.com/AWSCloudFormation/latest/UserGuide/aws-resource-efs-filesystem.html
+ *
+ * @resource AWS::EFS::FileSystem
+ */
+export class EfsFileSystem extends EFSFileSystemBase {
+
+    /**
+     * Import an existing File System from the given properties.
+     */
+    public static fromEfsFileSystemAttributes(scope: Construct, id: string, attrs: EfsFileSystemAttributes): IFileSystem {
```

Comment:
Can a filesystem imported in this way still be mounted? What would that look like?

Reply:
The mount script in the user data just needs the file systems Id, which we are setting in the object here. We additionally need to update the `connections` to allow connection for a different resource. 
Since we are setting both these properties here while importing an existing file system, it will be mounted using the same same way.